pilot/pkg/networking/core/v1alpha3/listener_builder.go

```diff
@@ -277,7 +277,7 @@ func (lb *ListenerBuilder) patchOneListener(l *listener.Listener, ctx networking
 }
 func (lb *ListenerBuilder) patchListeners() {
 	if lb.node.Type == model.Router {
-		envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_GATEWAY, lb.node, lb.push, lb.gatewayListeners, false)
+		lb.gatewayListeners = envoyfilter.ApplyListenerPatches(networking.EnvoyFilter_GATEWAY, lb.node, lb.push, lb.gatewayListeners, false)
```

Comment:
great, we can append listeners in `doListenerListOperation`. We should always set the return values.
Same applies to others (cluster, route), i have checked all others are ok